[docs] Add more details about concurrency on remote devices (#5804)
DevExpress/testcafe: docs/articles/documentation/guides/basic-guides/run-tests.md

* [Specify the Report Format](#specify-the-report-format)
* [Customize Screenshot and Video Settings](#customize-screenshot-and-video-settings)
* [Run Tests Concurrently](#run-tests-concurrently)
  * [Use Concurrency on Remote Devices](#use-concurrency-on-remote-devices)
* [Stage the Tested App](#stage-the-tested-app)

> * [Provide a Proxy URL](#provide-a-proxy-url)

Reviewer: 1) `* [Provide a Proxy URL](#provide-a-proxy-url)`

Can you rewrite this subtitle without using "Provide"?

2) I'm unable to edit line 38. Can you change it as follows?

`* [Run Tests from Multiple Sources](#run-tests-from-multiple-sources)`

Author: Okay, thank you, changed `Provide a Proxy URL` to `Specify a Proxy URL`

Reviewer: Thanks, Vasily.

> If an uncaught error or unhandled promise rejection occurs on the server during test execution, all tests running concurrently will fail.

### Use Concurrency on Remote Devices

When you run tests on [remote devices](../../reference/command-line-interface.md#remote-browsers), create browser connections for each instance of each browser you test against. The total number of instances is *c\*n*, where `c` is the concurrency factor and `n` is the number of browsers.

In the command line interface, specify the number of browser instances after the `remote:` keyword.  For example, use `-c 2 remote:6` to run tests in Chrome, Firefox, and Edge, with two instances of each browser. If you use the API, create a [browser connection](../../reference/testcafe-api/browserconnection/README.md) for each instance in code.

> Launch all the required browser instances manually on a remote device. The total number of instances should divide by the concurrency factor `c`. Otherwise, an exception is thrown.

Reviewer: `The total number of instances should divide by the concurrency factor `c`.`

Do you mean that the total number of instances are "divided by" the concurrency factor `c`? Please clarify.

Author: I would say (and @DIRECTcut also mentioned it in [his review](https://github.com/DevExpress/testcafe/pull/5804#discussion_r544361910)) "should be divisible by", but I remember getting a correction to "should divide by".

Does "should be divisible by" make sense?

Reviewer: "**x** should be divisible by **y**" means that when you divide **x** by **y**, the remainder is zero. For example, "4 is divisible by 2" is correct because when you divide 4 by 2, the remainder is zero.

Is this what you're trying to say here?

Author: Yes, that's what I mean. Thank you, I'll use "should be divisible" then, right?

Reviewer: Yes, you can use "should be divisible by" in this context.

```sh
testcafe -c 2 remote:4 tests/test.js
```

If you test against multiple remote browsers, open and connect all instances of one browser before you connect the next browser.

## Stage the Tested App

> 

Reviewer: Since I'm unable to edit line 648 (below), can you please update it as follows:

Line 648:
`1. A test runs for the first time. If it passes, TestCafe proceeds to the next test.`

Author: Thank you, done

Reviewer: Thank you!

> If an uncaught error or unhandled promise rejection occurs on the server during test execution, all tests running concurrently will fail.

### Use Concurrency on Remote Devices

When you run tests on [remote devices](../../reference/command-line-interface.md#remote-browsers),

> create browser connections for each instance of each browser you test against. The total number of instances is *c\*n*, where `c` is the concurrency factor and `n` is the number of browsers.

Reviewer 2: ```suggestion
create browser connections for each browser instance you test against. The total number of instances is *c\*n*, where `c` is the concurrency factor and `n` is the number of browsers.
```

> In the command line interface, specify the number of browser instances after the `remote:` keyword.  For example, use `-c 2 remote:6` to run tests in Chrome, Firefox, and Edge, with two instances for each browser. If you use the API, create a [browser connection](../../reference/testcafe-api/browserconnection/README.md) for each instance in code.

Reviewer 2: ```suggestion
In the command line interface, specify the number of browser instances after the `remote:` keyword.  For example, use `-c 2 remote:6` to run tests in Chrome, Firefox, and Edge, with two instances of each browser. If you use the API, create a [browser connection](../../reference/testcafe-api/browserconnection/README.md) for each instance in code.
```

Launch all the required browser instances manually on a remote device. The total number of instances

> should divide by the concurrency factor `c`. Otherwise, an exception is thrown.

Reviewer 2: You mean it should be divisible by a factor (without a remainder)? I'm not sure `should divide by` is the correct use. I might be wrong though...

```suggestion
should be divisible by the concurrency factor `c`. Otherwise, an exception is thrown.
```

Author: Yep, I also wrote "should be divisible" at first but got a correction to "should divide"

Reviewer: Please see my explanation below regarding "should be divisible".
